ql/src/java/org/apache/hadoop/hive/ql/exec/Utilities.java

```diff
@@ -2862,29 +2862,29 @@ public boolean skipProcessing(Task<?> task) {
    * corresponding to these dynamic partitions.
    */
   public static Map<Path, PartitionDetails> getFullDPSpecs(Configuration conf, DynamicPartitionCtx dpCtx,
-      Set<String> dynamicPartitionSpecs) throws HiveException {
+      Map<String, List<Path>> dynamicPartitionSpecs) throws HiveException {
 
     try {
       Path loadPath = dpCtx.getRootPath();
       FileSystem fs = loadPath.getFileSystem(conf);
       int numDPCols = dpCtx.getNumDPCols();
-      List<Path> allPartition = new ArrayList<>();
+      Map<Path, List<Path>> allPartition = new HashMap<>();
       if (dynamicPartitionSpecs != null) {
-        for (String partSpec : dynamicPartitionSpecs) {
-          allPartition.add(new Path(loadPath, partSpec));
+        for (Map.Entry<String, List<Path>> partSpec : dynamicPartitionSpecs.entrySet()) {
+          allPartition.put(new Path(loadPath, partSpec.getKey()), partSpec.getValue());
         }
       } else {
         List<FileStatus> status = HiveStatsUtils.getFileStatusRecurse(loadPath, numDPCols, fs);
         for (FileStatus fileStatus : status) {
-          allPartition.add(fileStatus.getPath());
+          allPartition.put(fileStatus.getPath(), null);
```

Comment:
Would it require much of refactor if we switch to Optional here?

Reply:
done